agot-bg-game-server/src/common/ingame-game-state/game-data-structure/westeros-card/WinterIsComingWesterosCardType.ts

```diff
@@ -12,16 +12,17 @@ export default class WinterIsComingWesterosCardType extends WesterosCardType {
             westerosDeckI: currentDeckI
         });
 
-        const deck = westerosGameState.game.westerosDecks[currentDeckI];
+        let deck = westerosGameState.game.westerosDecks[currentDeckI];
+        const dontShuffle = westerosGameState.game.revealedWesterosCards - 1;
 
         // Reset discarded state
         deck.forEach(card => { card.discarded = false; });
 
         // Shuffle the deck
-        shuffle(deck);
-
+        deck = deck.slice(0, dontShuffle).concat(shuffle(deck.slice(dontShuffle)));
```

Comment:
The last part `shuffle(deck.slice(dontShuffle))` will break the normal mode as `slice(-1)` is only returning the last card in the deck. I like the idea to solve it by a one liner but maybe an if condition makes life easier here.